bears/general/HTTPSBear.py

```diff
@@ -0,0 +1,75 @@
+from urllib.parse import urlparse
+
+from bears.general.URLBear import URLBear
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+
+from coalib.results.Diff import Diff
+from coalib.settings.Setting import typed_dict
+
+
+class HTTPSBear(LocalBear):
+    DEFAULT_TIMEOUT = 15
+    LANGUAGES = {'All'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Documentation'}
+    BEAR_DEPS = {URLBear}
+
+    def run(self, filename, file, dependency_results=dict(),
+            network_timeout: typed_dict(str, int, DEFAULT_TIMEOUT)=dict()):
+        """
+        Find http links in any text file and check if the https version of
+        link is valid. If so, an option is provided for replacing it with
+        https.
+
+        An https link is considered valid if the server responds with a 2xx
+        code.
+
+        Warning: This bear will make HEAD requests to all URLs mentioned in
+        your codebase, which can potentially be destructive. As an example,
+        this bear would naively just visit the URL from a line that goes like
+        `do_not_ever_open = 'https://api.acme.inc/delete-all-data'` wiping out
+        all your data.
+
+        :param dependency_results:
+            Results given by ``URLBear``.
+        """
+        HTTPS_PREFIX = 'https'
+        HTTP_PREFIX = 'http'
+        for result in dependency_results.get(URLBear.name, []):
+            line_number, link, code, context = result.contents
+            if (not link.startswith(HTTPS_PREFIX)):
+                httpslink = HTTPS_PREFIX + link[len(HTTP_PREFIX):]
```

Comment:
what if the link has a prefix that's not `http`? Or if it doesn't have a prefix at all (relative links)?

Reply:
Then nothing should happen because the purpose of HTTPSBear was just to find those links whose http can be updated to https, I mean then it should be under the scope of URLBear to take care.